src/sinks/elasticsearch/config.rs

```diff
@@ -369,44 +375,57 @@ impl DataStreamConfig {
 #[typetag::serde(name = "elasticsearch")]
 impl SinkConfig for ElasticsearchConfig {
     async fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
-        let common = ElasticsearchCommon::parse_config(self).await?;
+        let commons = ElasticsearchCommon::parse_endpoints(self).await?;
+        let common = commons[0].clone();
 
-        let http_client = HttpClient::new(common.tls_settings.clone(), cx.proxy())?;
-        let batch_settings = self.batch.into_batcher_settings()?;
+        let client = HttpClient::new(common.tls_settings.clone(), cx.proxy())?;
 
         let request_limits = self
             .request
             .tower
             .unwrap_with(&TowerRequestConfig::default());
 
-        let http_request_builder = HttpRequestBuilder {
-            bulk_uri: common.bulk_uri.clone(),
-            http_request_config: self.request.clone(),
-            http_auth: common.http_auth.clone(),
-            query_params: common.query_params.clone(),
-            region: common.region.clone(),
-            compression: self.compression,
-            credentials_provider: common.aws_auth.clone(),
-        };
+        let stream = if commons.len() > 1 {
```

Comment:
It _looks_ like the logic could actually be `let service = if commons.len() > 1 { ...` , and then the last two lines are common so don't need to have both in each conditional branch:

```
let sink = ElasticsearchSink::new(&common, self, service)?;
let stream = VectorSink::from_event_streamsink(sink);

Reply:
Yea, just looks like that. Those services have different types so we are constructing up to stream since it has the same type regardless of the service.